specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2023-01-01/dataprotection.json

```diff
@@ -0,0 +1,7082 @@
+      "enum": [
+        "Invalid",
+        "High",
+        "Standard"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "RehydrationPriority"
+      }
+    },
+    "RestoreFilesTargetInfo": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/RestoreTargetInfoBase"
+        }
+      ],
+      "description": "Class encapsulating restore as files target parameters",
+      "properties": {
+        "targetDetails": {
+          "$ref": "#/definitions/TargetDetails",
+          "description": "Destination of RestoreAsFiles operation, when destination is not a datasource"
+        }
+      },
+      "required": [
+        "recoveryOption",
+        "targetDetails"
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "RestoreFilesTargetInfo"
+    },
+    "RestoreJobRecoveryPointDetails": {
+      "properties": {
+        "recoveryPointID": {
+          "type": "string"
+        },
+        "recoveryPointTime": {
+          "format": "date-time",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "RestoreTargetInfo": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/RestoreTargetInfoBase"
+        }
+      ],
+      "description": "Class encapsulating restore target parameters",
+      "properties": {
+        "datasourceInfo": {
+          "$ref": "#/definitions/Datasource",
+          "description": "Information of target DS"
+        },
+        "datasourceSetInfo": {
+          "$ref": "#/definitions/DatasourceSet",
+          "description": "Information of target DS Set"
+        },
+        "datasourceAuthCredentials": {
+          "$ref": "#/definitions/AuthCredentials",
+          "description": "Credentials to use to authenticate with data source provider."
+        }
+      },
+      "required": [
+        "datasourceInfo",
+        "recoveryOption"
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "RestoreTargetInfo"
+    },
+    "RestoreTargetInfoBase": {
+      "description": "Base class common to RestoreTargetInfo and RestoreFilesTargetInfo",
+      "discriminator": "objectType",
+      "properties": {
+        "objectType": {
+          "description": "Type of Datasource object, used to initialize the right inherited type",
+          "type": "string"
+        },
+        "recoveryOption": {
+          "description": "Recovery Option",
+          "enum": [
+            "FailIfExists"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "RecoveryOption"
+          }
+        },
+        "restoreLocation": {
+          "description": "Target Restore region",
+          "type": "string"
+        }
+      },
+      "required": [
+        "objectType",
+        "recoveryOption"
+      ],
+      "type": "object"
+    },
+    "ResourceMoveDetails": {
```

Comment:
I don't think RPs usually document the resource move API, since that's a contract between RP and ARM, from RP's POV, or a contract between customer and ARM from customer's POV? Is this a special case?

Reply:
Looks like it is a special case.